.babelrc.js

```diff
@@ -13,38 +13,37 @@ module.exports = {
   // We need to preserve comments as they are used by webpack for
   // naming chunks during code-splitting. The compression step during
   // bundling will remove them later.
-  "comments": true,
+  comments: true,
 
-  "presets": [
-    ["@babel/env", {
-      // `targets` property set via `.browserslistrc`
-      "useBuiltIns": process.env.NO_COREJS_POLYFILL ? false : "usage",
-      "corejs": 3,
-      "modules": process.env.BABEL_MODULES ? process.env.BABEL_MODULES === 'false' ? false : process.env.BABEL_MODULES : "commonjs" // babel's default is commonjs
-    }],
-    ["@babel/typescript", { isTSX: true, allExtensions: true }],
-    "@babel/react"
+  presets: [
+    [
+      '@babel/preset-env',
+      {
+        // `targets` property set via `.browserslistrc`
+        useBuiltIns: process.env.NO_COREJS_POLYFILL ? false : 'usage',
+        corejs: 3,
```

Comment:
Just confirmed, a warning pops up when this line isn't changed:

```
WARNING (@babel/preset-env): The `corejs` option only has an effect when the `useBuiltIns` option is not `false`
```

Reply:
> corejs: process.env.NO_COREJS_POLYFILL ? undefined : 3,

I'm not sure about changing `false` to `undefined` because there is no difference between them. According to the documentation, the default value for `useBuiltIns` is `false` (https://babeljs.io/docs/babel-preset-env#usebuiltins). I guess it is worth leaving `false` because using values that are expected by the field will help avoid surprise issues in the future.

> I think because it doesn't want the corejs version unless 'usage' is specified for useBuiltIns. Did you run into the same thing at all?

Yes, I get the same warning message. The reason for this is that we purposely do not include polyfill imports in our build (see https://github.com/opensearch-project/oui/blob/main/scripts/compile-oui.js#L190-L219). As far as I understand, this solution was inherited from here https://github.com/elastic/eui/pull/1982.

Reply:
Can any of you help me with why we need any polyfills?

Reply:
I checked the code of `babel/preset-env` and @BSFishy's solution is a valid one.